components/uniset/src/provider.rs

```diff
@@ -14,78 +14,633 @@ pub mod key {
+    pub const TRAIL_CANONICAL_COMBINING_CLASS_VIRAMA_V1: ResourceKey = resource_key!(uniset, "20=9", 1);
+    pub const TRAIL_CANONICAL_COMBINING_CLASS_CCC91_V1: ResourceKey = resource_key!(uniset, "20=91", 1);
+    pub const VERTICAL_ORIENTATION_ROTATED_V1: ResourceKey = resource_key!(uniset, "21=0", 1);
+    pub const VERTICAL_ORIENTATION_TRANSFORMED_ROTATED_V1: ResourceKey = resource_key!(uniset, "21=1", 1);
+    pub const VERTICAL_ORIENTATION_TRANSFORMED_UPRIGHT_V1: ResourceKey = resource_key!(uniset, "21=2", 1);
+    pub const VERTICAL_ORIENTATION_UPRIGHT_V1: ResourceKey = resource_key!(uniset, "21=3", 1);
+    pub const WORD_BREAK_CR_V1: ResourceKey = resource_key!(uniset, "22=0", 1);
+    pub const WORD_BREAK_DOUBLE_QUOTE_V1: ResourceKey = resource_key!(uniset, "22=1", 1);
+    pub const WORD_BREAK_E_BASE_V1: ResourceKey = resource_key!(uniset, "22=2", 1);
+    pub const WORD_BREAK_E_BASE_GAZ_V1: ResourceKey = resource_key!(uniset, "22=3", 1);
+    pub const WORD_BREAK_E_MODIFIER_V1: ResourceKey = resource_key!(uniset, "22=4", 1);
+    pub const WORD_BREAK_EXTENDNUMLET_V1: ResourceKey = resource_key!(uniset, "22=5", 1);
+    pub const WORD_BREAK_EXTEND_V1: ResourceKey = resource_key!(uniset, "22=6", 1);
+    pub const WORD_BREAK_FORMAT_V1: ResourceKey = resource_key!(uniset, "22=7", 1);
+    pub const WORD_BREAK_GLUE_AFTER_ZWJ_V1: ResourceKey = resource_key!(uniset, "22=8", 1);
+    pub const WORD_BREAK_HEBREW_LETTER_V1: ResourceKey = resource_key!(uniset, "22=9", 1);
+    pub const WORD_BREAK_KATAKANA_V1: ResourceKey = resource_key!(uniset, "22=10", 1);
+    pub const WORD_BREAK_ALETTER_V1: ResourceKey = resource_key!(uniset, "22=11", 1);
+    pub const WORD_BREAK_LF_V1: ResourceKey = resource_key!(uniset, "22=12", 1);
+    pub const WORD_BREAK_MIDNUMLET_V1: ResourceKey = resource_key!(uniset, "22=13", 1);
+    pub const WORD_BREAK_MIDLETTER_V1: ResourceKey = resource_key!(uniset, "22=14", 1);
+    pub const WORD_BREAK_MIDNUM_V1: ResourceKey = resource_key!(uniset, "22=15", 1);
+    pub const WORD_BREAK_NEWLINE_V1: ResourceKey = resource_key!(uniset, "22=16", 1);
+    pub const WORD_BREAK_NUMERIC_V1: ResourceKey = resource_key!(uniset, "22=17", 1);
+    pub const WORD_BREAK_REGIONAL_INDICATOR_V1: ResourceKey = resource_key!(uniset, "22=18", 1);
+    pub const WORD_BREAK_SINGLE_QUOTE_V1: ResourceKey = resource_key!(uniset, "22=19", 1);
+    pub const WORD_BREAK_WSEGSPACE_V1: ResourceKey = resource_key!(uniset, "22=20", 1);
+    pub const WORD_BREAK_OTHER_V1: ResourceKey = resource_key!(uniset, "22=21", 1);
+    pub const WORD_BREAK_ZWJ_V1: ResourceKey = resource_key!(uniset, "22=22", 1);
 }
 
 #[derive(Debug, PartialEq, Clone)]
-#[cfg_attr(
-    feature = "provider_serde",
-    derive(serde::Serialize, serde::Deserialize)
-)]
+#[cfg_attr(feature = "provider_serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "testing", derive(Hash, Eq))]
```

Comment:
Based on feedback from fellow Rustaceans, please just unconditionally implement Hash and Eq, and remove the "testing" feature.

Reply:
Done.